### PR TITLE
:sparkles: Support setting maxHealthyPercentage to configure ASG instance refresh

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/31305

Backport of https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/5140

### CODE REVIEW

File: exp/api/v1beta2/awsmachinepool_webhook_test.go [hunk 1]
```diff
@@ -153,6 +153,27 @@ func TestAWSMachinePoolValidateCreate(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
```
Comment (mnitchev) [inline exp/api/v1beta2/awsmachinepool_webhook_test.go:156]: Maybe a test for when the value is above 200?

Comment (mnitchev) [reply exp/api/v1beta2/awsmachinepool_webhook_test.go:156]: and below 100

**Review by mnitchev — APPROVED** [review]
Minor stuff. Also if you add emojis to the title it should pass some of those checks

File: exp/api/v1beta2/awsmachinepool_webhook_test.go (hunk 1 above)
Comment (fiunchinho) [reply exp/api/v1beta2/awsmachinepool_webhook_test.go:156]: This is just cherry picking from upstream, so I'd prefer not to change the tests just in the fork.

**Pull request merged**